pyne/mesh.py

```diff
@@ -704,6 +704,16 @@ def __init__(self, mesh=None, structured=False,
             stored in self.dims.
 
         """
+        # if Mesh is made and no parameters passed, raise MeshError
+        if mesh is None and not structured and structured_coords is None and \
```

Comment:
This is a pretty beefy conditional statement and could be kinda hard to follow for future developers (though the comment helps a lot. Can I suggest making a tuple of our default args that will trigger this exception? Something like?

```python
defaults = (None, False, None, None, 'xyz', ())

if (mesh, structured, structured_coords, structured_set, structured_ordering) == defaults:
    raise MeshError(...)
```

Reply:
(Note that line may need to be wrapped to comply with PEP8)

Reply:
This seems wise. Are you willing to implement this suggestion @Mongwell ?

Reply:
An alternative may be to add parentheses to the original conditional to make it easier to parse.  I'm not 100% sold on @pshriwise solution because it kind of separates the values from their semantics - it's hard to follow what all the values in `defaults` refer to.

Reply:
Sure, I can see how that might be the case. It also isn't super flexible if the constructor changes in the future. I personally like the suggestion in [this PR](https://github.com/aruzzo/pyne/pull/1) though it goes perhaps even farther from the more verbose-yet-clear option here.